mlir/include/mlir/Dialect/EmitC/IR/EmitC.td

```diff
@@ -246,6 +247,81 @@ def EmitC_DivOp : EmitC_BinaryOp<"div", []> {
   let results = (outs FloatIntegerIndexOrOpaqueType);
 }
 
+def EmitC_ForOp : EmitC_Op<"for",
+      [AutomaticAllocationScope, DeclareOpInterfaceMethods<LoopLikeOpInterface,
```

Comment:
I see these as an output format, I'd not expect much optimization on them and given these represent the syntax for easy export more than anything, which could be difficult to encode in LoopLike interface.

Reply:
Agreed, better let interfaces be added if and when needed. Removing.